### PR TITLE
Add New Ruby-Saml vulnerability

### PR DESCRIPTION
**Pull request by VanessaHenderson**
This ruby-saml vuln came through the mailing lists this morning. Its not on OSVDB as far as I can tell (and I can't figure out how to create one) but a CVE was requested so for now I just put the date as the file name.

### CODE REVIEW

**Comment by phillmv** [comment]
CVEs are usually pretty quick on the uptake. Why not wait until later on/tomorrow?

**Comment by VanessaHenderson** [comment]
Okay, I'll see if they respond by tomorrow, I'll just let this sit in a PR until then :)

**Comment by phillmv** [comment]
:+1: thanks Vanessa!

**Comment by postmodern** [comment]
I will also adjust the specs to allow missing CVEs.

**Comment by VanessaHenderson** [comment]
What should the file names be in the case that there is no CVE/OSVDB? dates will get tedious, not that I'm expecting there to be many of these

**Comment by postmodern** [comment]
Ah, I see now, it's missing both CVE and OSVDB. Also email `moderators@osvdb.org` and see who assigns an ID first.

**Comment by postmodern** [comment]
That's a good question. Generally either MITRE or OSVDB would quickly assign an ID.

**Comment by phillmv** [comment]
So far the way we've been setup requires one or the other - and lately we've been preferring CVEs.

**Comment by reedloden** [comment]
I was the one who put in the requests to MITRE / oss-security and OSVDB... OSVDB got back to me and said they are experiencing infrastructure problems and aren't sure when they will be able to assign IDs again. MITRE takes forever for CVE requests.

As soon as I have an update, I'll let you know.

**Comment by reedloden** [comment]
OSVDB has assigned the XXE vuln (https://github.com/onelogin/ruby-saml/pull/247) as OSVDB-124383, though it won't show up on the site until they fix some problems. At least this unblocks us from getting this added. Still waiting for the assignment for the other vuln.

**Comment by VanessaHenderson** [comment]
I've added that one in there @reedloden :)

**Comment by reedloden** [comment]
and OSVDB has assigned the XPath injection (https://github.com/onelogin/ruby-saml/pull/225) as OSVDB-124991.

File: gems/ruby-saml/07092015_2.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+gem: ruby-saml
+cve: TBD
+url: https://github.com/onelogin/ruby-saml/pull/225
+title: Ruby-Saml Gem is vulnerable to XPath Injection
+date: 2015-07-09
+description: |
+  ruby-saml before 1.0.0 is vulnerable to XPath injection on xml_security.rb. The lack of prepared statements allows for possibly command injection, leading to arbitrary code execution
```
Comment (reedloden) [inline gems/ruby-saml/07092015_2.yml]: Can you wrap at 80 chars here?

**Comment by reedloden** [comment]
I think once you update 07092015_2 to reference OSVDB-124991 instead, this should be good to go.

**Comment by VanessaHenderson** [comment]
I was in the process of doing that!

File: gems/ruby-saml/OSVDB-124991.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+gem: ruby-saml
+osvdb: 124991
+url: https://github.com/onelogin/ruby-saml/pull/225
+title: Ruby-Saml Gem is vulnerable to XPath Injection
+date: 2015-07-09
```
Comment (reedloden) [inline gems/ruby-saml/OSVDB-124991.yml]: Let's make this 2015-04-29, as that's when https://github.com/onelogin/ruby-saml/commit/9853651b96b99653ea8627d757d46bfe62ab6448 was committed.

File: gems/ruby-saml/OSVDB-124383.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+gem: ruby-saml
+osvdb: 124383
+url: https://github.com/onelogin/ruby-saml/pull/247
+title: Ruby-Saml Gem is vulnerable to entity expansion attacks
+date: 2015-07-09
```
Comment (reedloden) [inline gems/ruby-saml/OSVDB-124383.yml]: Let's make this 2015-06-30, as that's when https://github.com/onelogin/ruby-saml/commit/a2e5318530701bf14528c5b3b51c880b3499a75d was committed.

**Comment by reedloden** [comment]
Awesome. Let's get the dates changed and squash your commits into one, and this should be good to go. :)

**Comment by reedloden** [comment]
Can you squash your commits into one? That's the only thing I see outstanding...

**Comment by postmodern** [comment]
:+1: I'll squash it for you.

**Comment by postmodern** [comment]
Merged by hand in d9ec6d2aa3984b969c9058d5277a3ab69d0e038d. Thank you for being patient!

**Pull request closed without merging**